### DISCUSSION TITLE
Upload in parts due to proxy

### DISCUSSION THREAD

**Discussion by Icenova20** [rejetto/hfs#447] [Q&A]
I'm proxying my connection to the server through Cloudflare. Cloudflare only supports uploading files over http/https up to 100MB per request.

Is the any way to upload files in parts, or in separate http requests, to circumvent this kind of limitation?

**Comment by rejetto** [comment]
hfs doesn't split the upload in parts, but it supports resuming if the upload is interrupted.
does cloudflare interrupt your upload, or doesn't block it at the start?

**Comment by Icenova20** [comment]
in the front end UI it appears to upload to 100% and then upon reaching 100% an error message pops up saying "Couldn't upload : file too large"

the partial hfs$ file does not remain in the filesystem

**Comment by rejetto** [comment]
can you see if in Admin-panel>monitoring you see the upload progressing?

**Comment by Icenova20** [comment]
It shows no progress on successful or failed uploads.

once the front end get to 100% on a successful upload the monitoring tab then shows that a file was uploaded, but does not display the upload progress.
on a failed upload the monitoring tab just shows a new connection but never shows any progress or that a file was uploaded.

**Comment by rejetto** [comment]
to use cloudflare, are you running a specific software on your computer or on your LAN?

**Comment by Icenova20** [comment]
That is an option, but currently, i am not running any software on my computer. The proxying is being done through DNS. My hostname points to cloudflare's servers and then they proxy that traffic to my IP.

**Comment by rejetto** [comment]
i was asking because since you don't see the progress, it must mean that it happens very quickly, like within 1 second.

Anyway, your idea of splitting the upload is technically possible, but not very easy and applies only to cloudflare. I don't expect to work on such a thing anytime soon.

**Comment by Icenova20** [comment]
I understand. Thank you for your consideration.
